Makefile

```diff
@@ -19,50 +26,48 @@ $(error "error Unexpected OS; Only Linux or Darwin supported.")
 endif
 endif
 
-# Environment
-SHELL := /bin/bash
-BASE_PATH := $(shell pwd)
-PUBLIC_PATH := $(BASE_PATH)/public
-THEMES_PATH := $(BASE_PATH)/themes
-STATIC_PATH := $(BASE_PATH)/static
-THEME_NAME := $(shell basename $(HUGO_THEME))
-THEME_PATH := $(THEMES_PATH)/$(THEME_NAME)
-HUGO_PATH := $(BASE_PATH)/.hugo
-HUGO_URL = https://github.com/spf13/hugo/releases/download/v$(HUGO_VERSION)/$(HUGO_TAR_FILE_NAME)
-HUGO_NAME := hugo_$(HUGO_VERSION)_$(ARCH)_$(OS)
-HUGO_URL_NAME := hugo_$(HUGO_VERSION)_$(URL_ARCH)-$(URL_OS)
-
 # Tools
-CURL = curl -L
-HUGO = $(HUGO_PATH)/hugo
-MKDIR = mkdir -p
-GIT = git
+CURL := curl -L
+HUGO := $(HUGO_PATH)/hugo
+MKDIR := mkdir -p
+JS_PACKAGE_MANAGER := yarn
 UNCOMPRESS := tar -zxf
 
 # Rules
 all: build
 
-init:
-	@if [ "$(HUGO_THEME)" == "" ]; then \
-		echo "ERROR! Please set the env variable 'HUGO_THEME' (http://mcuadros.github.io/autohugo/documentation/working-with-autohugo/)"; \
-		exit 1; \
-	fi;
+# Ensures the Hugo binary existance
+$(HUGO): $(HUGO_TAR_PATH)
+	$(UNCOMPRESS) $(HUGO_TAR_PATH) --directory=$(HUGO_PATH);
+
+# Downloads the hugo binary
+$(HUGO_TAR_PATH):
+	echo "Downloading $(HUGO_URL)"
+	$(MKDIR) $(HUGO_PATH)
+	$(CURL) $(HUGO_URL) -o $(HUGO_TAR_PATH)
 
-dependencies: init
-	@if [[ ! -f $(HUGO) ]]; then \
-		$(MKDIR) $(HUGO_PATH); \
-		cd $(HUGO_PATH); \
-		echo "Downloading $(HUGO_URL)"; \
-		$(CURL) $(HUGO_URL) -o $(HUGO_TAR_FILE_NAME); \
-		$(UNCOMPRESS) $(HUGO_TAR_FILE_NAME); \
-	fi;
+# Ensures hugo dependencies
+hugo-dependencies: $(HUGO_TAR_PATH) $(HUGO)
 
-build: dependencies
-	$(HUGO) -t $(THEME_NAME) --baseURL $(BASE_URL)
+# Prepares yarn
+js-dependencies:
+	$(JS_PACKAGE_MANAGER) install --force
+	$(JS_PACKAGE_MANAGER) build
+
+# Prepares project dependencies
+project-dependencies: hugo-dependencies js-dependencies
```

Comment:
Why this cannot be called `dependencies` as before?

Reply:
This rulename was brought from the [landing makefile](https://github.com/src-d/landing/blob/master/Makefile#L100) (pretending to do the same -at least, as much as possible- in our different web apps).
In the landing it has not been called only `dependencies` because that name is already used by the [ci/Makefile.main](https://github.com/src-d/ci/blob/master/Makefile.main) -that could be used here in a near future-.

Reply:
Well, if this could inherit from that `Makefile` it would. If it cannot inherit from it as it works differently, it should be clear that we're overriding default `source{d}` behaviour

Reply:
Since having a Makefile as much similar as the Landing one would be great,
would it solve your requirement?

Reply:
I don't think the status of landing `Makefile` is ideal... It's unfortunate that it does not follow the company conventions. It's weird to have some projects with a `dependencies` action and others with `project-dependencies` as I don't really grasp the difference.

Company-wide Makefile `dependencies` action looks very focused towards building go. For me it would make total sense to override `dependencies` here even if we ever inherit from it as we're not building go.

Anyway, I'm not maintainer from this repo, so I shouldn't have final word on this.

Reply:
Agree with that. Thanks for the clear explanation.
I opened an issue to face it as soon as possible https://github.com/src-d/blog/issues/150

Reply:
Agreed with @dpordomingo that this will follow landing conventions for the moment until we unify all platform Makefiles